src/toil/cwl/cwltoil.py

```diff
@@ -1190,11 +1204,41 @@ def run(self, file_store):
     "StepInputExpressionRequirement", "ResourceRequirement")
 
 
-def visitSteps(t, op):
-    if isinstance(t, cwltool.workflow.Workflow):
-        for s in t.steps:
-            op(s.tool)
-            visitSteps(s.embedded_tool, op)
+def visitSteps(cmdline_tool, op):
+    # : Union[cwltool.command_line_tool.CommandLineTool, cwltool.workflow.Workflow]
+    if isinstance(cmdline_tool, cwltool.workflow.Workflow):
+        for step in cmdline_tool.steps:
+            op(step.tool)
+            visitSteps(step.embedded_tool, op)
+
+
+def remove_unprocessed_secondary_files(unfiltered_secondary_files):
+    """
+    Interpolated strings and optional inputs in secondary files were added to CWL in version 1.1.
+
+    The CWL libraries we call do successfully resolve the interpolated strings, but add the resolved
+    fields to the list of unresolved fields so we remove them here after the fact.
+
+    We also remove any secondary files here not containing 'toilfs:', which means that it was not
+    successfully imported into the toil jobstore.  The 'required' logic seems to be handled deeper in
+    cwltool.builder.Builder(), and correctly determines which files should be imported.  Therefore we
+    remove the files here and if this file is SUPPOSED to exist, it will still give the appropriate file
+    does not exist error, but just a bit further down the track.
+    """
+    intermediate_secondary_files = []
+    final_secondary_files = []
+    # remove secondary files still containing interpolated strings
+    for sf in unfiltered_secondary_files['secondaryFiles']:
+        sf_bn = sf.get('basename', '')
+        if ('$(' not in sf_bn) and ('${' not in sf_bn):
```

Comment:
There's no way in CWL to escape these byte pairs in a file name is there? Or have them substituted in later? Because I can definitely have a file on disk called '${', and I might try and run a CWL workflow on it just to upset people.

Reply:
I agree.  I made sure to use cwltool's method of checking, however, (https://github.com/common-workflow-language/cwltool/blob/master/cwltool/builder.py#L383 ... and several other places), since I wanted to make sure that whatever issues we had, they should also have.

I think they just rely on users not having files named '${', though thinking about it, I can't think of a good way to allow string interpolation AND allow files/dirs with interpolation-y names.